docs/v1/api.md

```diff
@@ -1,63 +1,96 @@
+created to be consumed by an application.
 
 This resource is created by the service catalog's controller event loop after
 it has received a `Broker` resource and successfully called the backing CF
-broker's catalog endpoint. It represents a single (serviceID, planID) pair that
-a CF service broker has made available.
+broker's catalog endpoint.
+
+*TODO: Make explicit the relationship between service classes and plans*
+
+## `Instance` Resource
+
+An `Instance` represents a provisioned instance of a `ServiceClass`, and is the
+entity an application binds to.
+
+A service consumer creates an `Instance` to indicate their intent to provision
+an instance of a service class.  The `Instance` has a reference to the
+`ServiceClass` to provision an instance of.
+
+The service catalog controller has a watch on the `Instance` resource and
+receives an `ADD` watch event. The controller then attempts to provision a new
+instance:
 
-*TODO: what happens in the below cases?*
+1.  The controller calls the provision endpoint on the broker 
+2.  The broker returns a response indicating that the broker provisioned the new
+    instance, the instance was already provisioned, or the provisioning
+    operation is in progress
+3.  The controller updates the status of the `Instance` to indicate when the
+    `Instance` is in a provisioning or provisioned condition
 
-1. *The `Broker` that caused the `ServiceClass` was deleted*
-1. *The `ServiceClass` itself was deleted*
+*TODO: codify how asynchronous responses are handled in the controller*
 
-## `Instance`
+## `Binding` Resource
 
-This resource is created by a service consumer to indicate their desire to
-provision a service. When the service catalog's controller event loop sees an
-`Instance` created, it calls the provision endpoint on the backing CF service
-broker and writes `provisioned` into a `status` field of the
-`Instance`.
+A `Binding` represents the binding between an application and an `Instance` of a
+`ServiceClass`.  
 
-*TODO: what happens when an `Instance` resource is deleted?*
+*TODO: clarify exactly what constitutes an application.*
 
-## `Binding`
+Service consumers create `Binding` resources to indicate that an application
+should be bound to an `Instance`.  The `Binding` contains information about how
+the application wants to use the binding information such as:
 
-This resource is created by a service consumer to indicate that an application
-should be bound to an instance. When the service catalog's controller event
-loop sees a `Binding` created, it calls the bind endpoint on the backing CF
-service broker. When a successful response is returned, the event loop does
-the following:
+1.  The name of a Kubernetes core `Service` resource to provide a stable
+    endpoint for the application to use the `Instance` via
+2.  The name of a Kubernetes `Secret` resource to hold credentials necessary to
+    use the service
```

Comment:
Perhaps add something like:
If these values are not provided then the name of the service instance will be used by default.